### PR TITLE
Fix CORE-LogCollector.sh for multiple distros

### PR DESCRIPTION
**Pull request by asofron**
Output after fix : 

**Centos:**
[azure] 12/14/2018 07:57:41 : [INFO ] Collecting Operating System Logs.....
[azure] \S
[azure] Kernel \r on an \m
[azure] Operating system Log process finished...
[azure] Collecting Waagent Details....
[azure] Collecting Microsoft Linux Integration Service Data...
[azure] hv_netvsc module: 3.10.0-862.11.6.el7.x86_64
[azure] hv_storvsc module: 3.10.0-862.11.6.el7.x86_64
[azure] scsi_transport_fc module: 3.10.0-862.11.6.el7.x86_64
[azure] hv_vmbus module: 3.10.0-862.11.6.el7.x86_64
[azure] Collecting Microsoft Linux Integration Service Data Finished...
[azure] Collecting Disk and Memory Data
[azure] Collecting Disk and Memory Data Finished...
[azure] Collecting Processor Data...
[azure] Collecting Processor Data Finished...
[azure] Collecting Network Data...
[azure] Collecting Network Data Finished...
[azure] Compressing Logs

**SLES:**
[azure] 12/14/2018 08:48:31 : [INFO ] Collecting Operating System Logs.....
[azure] Collecting Waagent Details....
[azure] Collecting Microsoft Linux Integration Service Data...
[azure] Info: Skipping hv_vmbus module as it is built-in.
[azure] Info: Skipping hv_storvsc module as it is built-in.
[azure] All modules built-in ...
[azure] Collecting Microsoft Linux Integration Service Data Finished...
[azure] Collecting Disk and Memory Data
[azure] Collecting Disk and Memory Data Finished...
[azure] Collecting Processor Data...
[azure] Collecting Processor Data Finished...
[azure] Collecting Network Data...
[azure] netstat is not installed
[azure] route is not installed
[azure] Compressing **Logs**

**Ubuntu :** 

[azure] 12/14/2018 08:59:37 : [INFO ] Collecting Operating System Logs.....
[azure] Ubuntu 18.04.1 LTS \n \l
[azure] Operating system Log process finished...
[azure] Collecting Waagent Details....
[azure] Collecting Microsoft Linux Integration Service Data...
[azure] Info: Skipping hv_vmbus module as it is built-in.
[azure] Info: Skipping hv_storvsc module as it is built-in.
[azure] hv_netvsc module: 4.15.0-1035-azure
[azure] Collecting Microsoft Linux Integration Service Data Finished...
[azure] Collecting Disk and Memory Data
[azure] Collecting Disk and Memory Data Finished...
[azure] Collecting Processor Data...
[azure] Collecting Processor Data Finished...
[azure] Collecting Network Data...
[azure] Collecting Network Data Finished...
[azure] Compressing Logs

### CODE REVIEW

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 1]
```diff
@@ -43,8 +43,12 @@ Collect_OS_Logs() {
     echo "Collecting Operating System Logs....."
     Log "Collecting Operating System Details at" 'date' $dirname/OS.log
     Log "Kernel Version" 'uname -a' $dirname/OS.log
+    if [ -f "/etc/issue" ]; then
     Log "Distro Release Details" 'cat /etc/issue' $dirname/OS.log
-    Log "Additional Kernel Details" 'cat /proc/version' $dirname/OS.log
+    else
```
Comment (simonxiaoss) [inline Testscripts/Linux/CORE-LogCollector.sh]: Indent?

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 2]
```diff
@@ -76,8 +80,40 @@ Collect_OS_Logs() {
...
+                break
+            fi
+        done
+        if ${KEEP}; then
+            TEMP_HYPERV_MODULES+=(${mod})
+        fi
+    done
+    HYPERV_MODULES=("${TEMP_HYPERV_MODULES[@]}")
+    if [ -z "$HYPERV_MODULES" ];then
+    echo "No modules loaded..."
```
Comment (simonxiaoss) [inline Testscripts/Linux/CORE-LogCollector.sh]: indent

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 3]
```diff
@@ -111,11 +147,21 @@ Collect_Processor() {
 }
 
 Collect_Network() {
+    ifconfig=$(command -v ifconfig)
+    netstat=$(command -v netstat)
     echo "Collecting Network Data..."
+    if ! [ "$ifconfig" ]; then
+    echo "ifconfig is not installed"
```
Comment (simonxiaoss) [inline Testscripts/Linux/CORE-LogCollector.sh]: indent

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 4]
```diff
@@ -76,8 +80,40 @@ Collect_OS_Logs() {
...
     echo "Collecting Microsoft Linux Integration Service Data..."
-    vsc_modules=$(lsmod | grep vsc | cut -d' ' -f1)
-    for module in "${vsc_modules[@]}"; do
+    HYPERV_MODULES=()
+    HYPERV_MODULES+=$(lsmod | grep vsc | cut -d' ' -f1)
+    skip_modules=()
+    vmbus_included=`grep CONFIG_HYPERV=m /boot/config-$(uname -r)`
+    if [ $vmbus_included ]; then
+        skip_modules+=("hv_vmbus")
+        echo "Info: Skiping hv_vmbus module as it is built-in."
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/CORE-LogCollector.sh]: Using Log?

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 5]
```diff
@@ -76,8 +80,40 @@ Collect_OS_Logs() {
...
+    storvsc_included=`grep CONFIG_HYPERV_STORAGE=m /boot/config-$(uname -r)`
+    if [ $storvsc_included ]; then
+        skip_modules+=("hv_storvsc")
+        echo "Info: Skiping hv_storvsc module as it is built-in."
+    fi
+    # Remove each module in HYPERV_MODULES from skip_modules
+    TEMP_HYPERV_MODULES=()
+    for mod in "${HYPERV_MODULES[@]}"; do
+        for remove in "${skip_modules[@]}"; do
+            KEEP=true
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/CORE-LogCollector.sh]: move line between 97 and 98

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 6]
```diff
@@ -111,11 +147,21 @@ Collect_Processor() {
...
+    ifconfig=$(command -v ifconfig)
+    netstat=$(command -v netstat)
     echo "Collecting Network Data..."
+    if ! [ "$ifconfig" ]; then
+    echo "ifconfig is not installed"
+    else
     Log "Network Interface Details" 'ifconfig -a' $dirname/Network.txt
+    fi
+    if ! [ "$netstat" ]; then
+    echo "netstat is not installed"
```
Comment (juhlee-microsoft) [inline Testscripts/Linux/CORE-LogCollector.sh]: indentation here and below

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 1 above)
Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: Fixed

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 2 above)
Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: Fixed

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 3 above)
Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: Fixed

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 4 above)
Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: Log is function (was already in the script) that echo command used  and command output to a file no realtionship with LogMsg

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 6 above)
Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: Done

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 5 above)
Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: Done

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 7]
```diff
@@ -111,11 +147,21 @@ Collect_Processor() {
 }
 
 Collect_Network() {
+    ifconfig=$(command -v ifconfig)
```
Comment (chvalean) [inline Testscripts/Linux/CORE-LogCollector.sh]: replace all ifconfig use in this script with the "ip" command

Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: change to ip

File: Testscripts/Linux/CORE-LogCollector.sh (hunk 4 above)
Comment (ader1990) [reply Testscripts/Linux/CORE-LogCollector.sh]: Skipping* hv_vmbus as it is a built-in module

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 8]
```diff
@@ -76,8 +80,40 @@ Collect_OS_Logs() {
...
+    skip_modules=()
+    vmbus_included=`grep CONFIG_HYPERV=y /boot/config-$(uname -r)`
+    if [ $vmbus_included ]; then
+        skip_modules+=("hv_vmbus")
+        echo "Info: Skiping hv_vmbus module as it is built-in."
+    fi
+    storvsc_included=`grep CONFIG_HYPERV_STORAGE=y /boot/config-$(uname -r)`
+    if [ $storvsc_included ]; then
+        skip_modules+=("hv_storvsc")
+        echo "Info: Skiping hv_storvsc module as it is built-in."
```
Comment (ader1990) [inline Testscripts/Linux/CORE-LogCollector.sh]: ditto

File: Testscripts/Linux/CORE-LogCollector.sh [hunk 9]
```diff
@@ -76,8 +80,40 @@ Collect_OS_Logs() {
...
+                break
+            fi
+        done
+        if ${KEEP}; then
+            TEMP_HYPERV_MODULES+=(${mod})
+        fi
+    done
+    HYPERV_MODULES=("${TEMP_HYPERV_MODULES[@]}")
+    if [ -z "$HYPERV_MODULES" ];then
+        echo "No modules loaded..."
```
Comment (ader1990) [inline Testscripts/Linux/CORE-LogCollector.sh]: this is not an accurate information, you mean that the kernel modules are already built in

Comment (asofron) [reply Testscripts/Linux/CORE-LogCollector.sh]: FIxed

**Comment by chvalean** [comment]
@simonxiaoss @ader1990 please approve this so we can merge

**Pull request merged**